x-pack/plugins/enterprise_search/server/lib/ml_inference_pipeline/get_inference_pipelines.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core/server';
+import { MlTrainedModels } from '@kbn/ml-plugin/server';
+
+import { IngestPipeline, IngestProcessorContainer } from '@elastic/elasticsearch/lib/api/types';
+
+/**
+ * Gets all ML inference pipelines. Redacts trained model IDs in those pipelines which reference
+ * a model inaccessible in the current Kibana space.
+ * @param esClient the Elasticsearch Client to use to fetch the errors.
+ * @param trainedModelsProvider ML trained models provider.
+ */
+export const getMlInferencePipelines = async (
+  esClient: ElasticsearchClient,
+  trainedModelsProvider: MlTrainedModels | undefined,
+): Promise<Record<string, IngestPipeline>> => {
+  if (!trainedModelsProvider) {
+    return Promise.reject(new Error('Machine Learning is not enabled'));
+  }
+
+  // Fetch all ML inference pipelines
+  const fetchedInferencePipelines = await esClient.ingest.getPipeline({
+    id: 'ml-inference-*',
+  });
+
+  // Fetch all trained model IDs that are accessible in the current Kibana space
+  const trainedModels = await trainedModelsProvider.getTrainedModels({});
```

Comment:
```suggestion
  // Fetch all ML inference pipelines & trained model IDs that are accessible in the current Kibana space
  const [fetchedInferencePipelines, trainedModels] = await Promise.all([
    esClient.ingest.getPipeline({
      id: 'ml-inference-*',
    }),
    trainedModelsProvider.getTrainedModels({}),
  ]);
```

Probably not a huge difference but this would execute the two queries in parallel.

Reply:
Good idea, I updated the code.